src/waltham/waltham-connection.h

```diff
@@ -373,6 +354,83 @@ wth_connection_get_protocol_error(struct wth_connection *conn,
 				  const char **interface,
 				  uint32_t *object_id);
 
+/** Return a registry using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The registry obtained.
+ *
+ * The returned registry can then be used to retrieve other global
+ * objects, using its usual interface.
+ *
+ * \memberof wth_connection
+ * \client_api
+ */
+struct wthp_registry *
+wth_connection_get_registry(struct wth_connection *conn);
+
+/** Return a callback using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The callback obtained.
+ *
+ * The returned callback will emit a 'done' event once the related
+ * request is done.
+ *
+ * \memberof wth_connection
+ * \client_api
+ */
+struct wthp_callback *
+wth_connection_sync(struct wth_connection *conn);
+
+/** Set the callback to be called once a client requests a registry
+ *
+ * \param conn The Waltham connection.
+ * \param callback The registry callback.
```

Comment:
I wonder where we could document the function pointer better, as in, when it gets called what are the arguments to it. In Weston I've been using a documented `typedef` but as we don't generate Doxygen docs in Weston, I don't know if that actually works.

Reply:
This actually worked quite well. :)